PURL-TYPES.rst

```diff
@@ -59,6 +59,26 @@ apk
       pkg:apk/alpine/curl@7.83.0-r0?arch=x86
       pkg:apk/alpine/apk@2.12.9-r3?arch=x86
 
+bazel
+-----
+``bazel`` for Bazel modules as specified at ``https://bazel.build/external/module``:
+
+- The default repository is the Bazel Central Registry (BCR) ``https://bcr.bazel.build``.
+- The ``name`` is the module name.
```

Comment:
@fmeum In most case-insensitive types, we state this:
`The name is not case sensitive and must be lowercased.` 
@Yannic I reckon the (minor) hardship on type-specific cae normalization, but we already do it everywhere else, and this help using a PURL consistently as a key in databases.

```suggestion
- The ``name`` is the module name and must be lowercased.
```

Reply:
I pushed an update, is that what you had in mind? Please note that Bazel module names are not case insensitive in the literal sense, it's an error to request a module with a name that isn't lowercased. But I guess that applies to other schemes as well and normalizing them is simpler than marking them invalid?

Reply:
> @fmeum In most case-insensitive types, we state this: `The name is not case sensitive and must be lowercased.` @Yannic I reckon the (minor) hardship on type-specific cae normalization, but we already do it everywhere else, and this help using a PURL consistently as a key in databases.

I disagree. 
we'vediscussesd this already, andlearned, that the name's case is highlyimportant.
therefore, i woudl rather go with the completely opposite declaration:
> - The ``name`` is case sensitive and MUST NOT be lowercased.

Reply:
> The name is case sensitive and MUST NOT be modified.

I like that this explicitly doesn't mention what the casing should be (that's left to the domain), just that it is important.

Reply:
@jkowalleck I updated spec and test case, PTAL.

Reply:
> > The name is case sensitive and MUST NOT be modified.
> 
> I like that this explicitly doesn't mention what the casing should be (that's left to the domain), just that it is important.

exactly.



In the end, I would really prefer to not mention any normalization or rule here.
this pins purl in a place where it has no authority.
the package ecosystemsmayevolve and chaneg their rules, and wewould (again) end upwith incompatible purl types and endless discussions.

Reply:
bazil is excplic it about caseing: it is case-sensitive, and no normalization is implied.


"name must be lowercased" -> meaning name is case-sensitive and Uppercase is forbidden today. bazil does not know any normalization, and does at no point imply that it will never evolve.
bazil is the authority, not purl. dont make the same mistakes again and start normalization where none is planned.

to contrast this: pypi clearly states how normalization work.

Reply:
I agree with what @jkowalleck and @fmeum said above:

Bazel module names are case sensitive. `foo`, `FOO`, and `FoO` are all separate modules that could theoretically exist side by side. The fact that there are currently no such modules does not matter.

I agree with the part that `foo` and `FOO` are unlikely to ever exist at the same time, but I could very well see `foo` and `BAR` both existing with exactly this casing.

IMO, `purl` should uniquely identify a package, and I should be able to take the `name`, go to the package manager, and get to the right package. Normalization breaks this invariant, since `bar` from the example above is going to break this.

I like `The name is case sensitive and MUST NOT be modified.` as phrasing

Reply:
Note that different casings can also happen if private registries are involved: public BCR may happen to contain `foo`, which the private registry at $company mirrors, but there can also be an additional module `FOO` in the private registry